contrib/labstack/echo.v4/echotrace.go

```diff
@@ -52,12 +52,22 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 			// serve the request to the next middleware
 			err := next(c)
 			if err != nil {
-				span.SetTag(ext.Error, err)
-				// invokes the registered HTTP error handler
-				c.Error(err)
```

Comment:
Why is this call removed?

Reply:
This is what I was referring to above, about checking the result of the error handler.

After looking deeper into the echo internals, it turns out it's not safe to call `c.Error` like that, since it can have permanent affects on the response which preclude error handlers higher on the middleware stack from handling errors correctly.

In addition, the error handler may not be idempotent, meaning calling it here and also having echo call it a second time later (which it does) may cause problems. 

Finally, there's no way to ensure that the result set by calling `c.Error` is the true final result, since (again) middleware higher on the stack may affect the response and calling the error handler a second time may produce a different result. 

In short, it's not safe and we shouldn't have been doing it before.